### PR TITLE
get_wait_list and wait_for_all(HandleList)

### PR DESCRIPTION
**Pull request by AnuyaWelling2801**
Adding get_wait_list and wait_for_all for a list of SyncTypes. 
Changes to the concurrent queue to add a pop_all function.

### CODE REVIEW

**Review by vossmjp — CHANGES_REQUESTED** [review]
Need to rename to wait_for_all to wait. Also should make list type generic in free function. Maybe add more tests.

File: include/oneapi/dpl/internal/dynamic_selection.h [hunk 1]
```diff
@@ -71,6 +72,18 @@ namespace experimental {
     return std::forward<Handle>(h).wait_for_all();
   }
 
+  template<typename Handle>
+  auto wait_for_all(std::list<Handle> l) {
```
[the remark this answers is not in the document]
Comment (AnuyaWelling2801) [reply include/oneapi/dpl/internal/dynamic_selection.h]: Having std::list to differentiate between wait(Handle &h) v/s wait(HandleList &l) since there is no way to specify a single element vs a list without std::concepts

File: include/oneapi/dpl/internal/dynamic_selection_impl/concurrent_queue.h [hunk 2]
```diff
@@ -48,6 +49,21 @@ namespace util{
             cond_.notify_one();
         }
 
+        void pop_all(std::list<T>& item_list)
+        {
+            std::unique_lock<std::mutex> mlock(mutex_);
+            while (queue_.empty())
```
[the remark this answers is not in the document]
Comment (AnuyaWelling2801) [reply include/oneapi/dpl/internal/dynamic_selection_impl/concurrent_queue.h]: Changed

File: test/support/concurrent_queue.h [hunk 3]
```diff
@@ -45,6 +45,21 @@ namespace TestUtils{
             cond_.notify_one();
         }
 
+        void pop_all(std::list<T>& item_list)
+        {
+            std::unique_lock<std::mutex> mlock(mutex_);
+            while (queue_.empty())
```
[the remark this answers is not in the document]
Comment (AnuyaWelling2801) [reply test/support/concurrent_queue.h]: Changed

File: test/support/test_ds_utils.h [hunk 4]
```diff
@@ -82,6 +82,44 @@ int test_invoke_async_and_wait_on_policy(UniverseContainer u, ResourceFunction&&
   return 0;
 }
 
+template<typename Policy, typename UniverseContainer, typename ResourceFunction>
+int test_invoke_async_and_get_wait_list(UniverseContainer u, ResourceFunction&& f) {
```
[the remark this answers is not in the document]
Comment (AnuyaWelling2801) [reply test/support/test_ds_utils.h:86]: Added more tests for empty list and a single element
